addons/armaos/functions/fnc_terminal_addEventHandler.sqf

```diff
@@ -0,0 +1,168 @@
+/**
+ * Adds event handler to the user interface elements of a terminal.
+ *
+ * Arguments:
+ * 1: UI Text Field <CONTROL>
+ * 2: Keyboard Layout UI Button <CONTROL>
+ *
+ * Results:
+ * None
+ */
+
+#include "\a3\ui_f\hpp\definedikcodes.inc"
+
+params ["_terminalCtrl", "_languageButtonCtrl"];
+
+/* ================================================================================ */
+
+private _result = _terminalCtrl ctrlAddEventHandler
+[
+	"KeyDown", 
+	{
+		params ["_displayorcontrol", "_key", "_shift", "_ctrl", "_alt"];
+
+		private _computer = _displayorcontrol getVariable "AE3_computer";
+		private _terminal = _computer getVariable "AE3_terminal";
+		private _terminalApplication = _terminal get "AE3_terminalApplication";
+		private _terminalAllowedKeys = _terminal get "AE3_terminalAllowedKeys";
+		private _terminalBuffer = _terminal get "AE3_terminalBuffer";
+		private _terminalPrompt = _terminal get "AE3_terminalPrompt";
+
+		private _lastBufferLineIndex = (count _terminalBuffer) - 1;
+
+		_terminal set ["AE3_terminalScrollPosition", 0];
+
+		private _keyCombination = format ["%1-%2-%3-%4", _key, _shift, _ctrl, _alt];
+
+		/* ---------------------------------------- */
+
+		if (_keyCombination in _terminalAllowedKeys) then
+		{
+			private _keyChar = _terminalAllowedKeys get _keyCombination;
+
+			switch (_terminalApplication) do
```

Comment:
More efficient:
```sqf
[_terminalApplication, _computer, _keyChar] call
{
params ['_terminalApplication', '_computer', '_keyChar'];
if (_terminalApplication isEqualTo "LOGIN" exitWith { ... };
....
};

```

Reply:
I will change that.

Reply:
changed  multiple times in this file, see commit https://github.com/y0014984/Advanced-Equipment/pull/90/commits/351ca2f7548ca9717ff26f606317f2c7e56c0763